admin/provisioner/clickhousestatic/provisioner.go

```diff
@@ -53,32 +52,40 @@ func New(specJSON []byte, _ database.DB, logger *zap.Logger) (provisioner.Provis
 		return nil, fmt.Errorf("failed to parse provisioner spec: %w", err)
 	}
 
-	if spec.DSNPath != "" {
-		dsn, err := os.ReadFile(spec.DSNPath)
-		if err != nil {
-			return nil, fmt.Errorf("failed to read DSN file: %w", err)
-		}
-		spec.DSN = strings.TrimSpace(string(dsn))
+	// Get primary DSN from the specified environment variable (required)
+	dsn := os.Getenv(spec.DSNEnv)
+	if dsn == "" {
+		return nil, fmt.Errorf("environment variable %s is not set or empty", spec.DSNEnv)
```

Comment:
Use `%q` so errors due to empty strings are easier to notice
```suggestion
		return nil, fmt.Errorf("environment variable %q is not set or empty", spec.DSNEnv)
```